app/api-v1/controllers/Attachment/index.js

```diff
@@ -1,5 +1,79 @@
+const { parseAccept } = require('../../../utils/parseAcceptHeader')
+const logger = require('../../../utils/Logger')
+
+const db = require('../../../db')
+
+const { createAttachmentFromFile, createAttachment } = require('../Attachment/helpers')
+const { InternalError, NotFoundError, BadRequestError } = require('../../../utils/errors')
+
 module.exports = {
-  getAll: async function () {
-    return { status: 500, response: { message: 'Not Implemented' } }
+  get: async (req) => {
+    /*
+    if (orderedAccept.includes(jsonMimeTypes) && filename === 'json') {
+        return {
+          status: 200,
+          response: JSON.parse(JSON.stringify(attachment)),
+        }
+      }
+
+      return returnWithFile(attachment)
+  */
+    const { id } = req.params
+    if (!id) throw new BadRequestError('missing params')
+
+    const [attachment] = await db.getAttachment(req.params.id)
+    if (!attachment) throw new NotFoundError('Attachment Not Found')
+
+    const orderedAccept = parseAccept(req.headers.accept)
+
+    for (const mimeType of orderedAccept) {
+      if (mimeType === 'application/json' || mimeType === 'application/*' || mimeType === '*/*') {
+        try {
+          const json = JSON.parse(JSON.stringify(attachment.binary_blob))
+          return { status: 201, response: { binary_blob: json, filename: attachment.filename, id: attachment.id } }
+        } catch (error) {
+          throw new InternalError({ message: error })
+        }
+      }
+      if (mimeType === 'application/octet-stream') {
+        return {
+          status: 201,
+          response: {
+            headers: {
+              immutable: true,
+              maxAge: 365 * 24 * 60 * 60 * 1000,
+              'content-disposition': `attachment; filename="${attachment.filename}"`,
+              'access-control-expose-headers': 'content-disposition',
+              'content-type': 'application/octet-stream',
+            },
+            file: { binary_blob: attachment.binary_blob, filename: attachment.filename, id: attachment.id },
+          },
+        }
+      }
+    }
+    throw new InternalError({ message: 'Client file request not supported' })
```

Comment:
Glad to see a simplified version

1. try/catch is irelevant here. We either throw or allow to throw as we have a middle ware. Happy to walk through.
2. the logic is very hard to understand by just looking at it. I think it could be simplified, this will also mean that testing becomes more difficult
3. seems like we are using parseAccept which should do most of the heavy lifting when it comes to headers? I would leen toward moviing this logic into to a helper where it could tested in isolation for example


again please take this with pinch of salt and for as long as it works - it works.